modules/account-lib/src/coin/dot/iface.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseTxInfo, TypeRegistry, DecodedUnsignedTx } from '@substrate/txwrapper-core/lib/types';
+
+export interface Seed {
+  seed: Uint8Array;
+}
+export type specNameType = 'kusama' | 'polkadot' | 'westend' | 'statemint' | 'statemine';
+
+export interface TxData {
+  sender: string;
+  blockHash: string;
+  blockNumber: number;
+  genesisHash: string;
+  metadataRpc: string;
+  nonce: number;
+  specVersion: number;
+  transactionVersion: number;
+  chainName: string;
+  specName?: string;
+  amount?: string;
+  dest?: string;
+  tip?: number;
+  eraPeriod?: number;
+}
+
+export interface TransferArgs {
+  dest: { id: string };
+  value: string;
+}
+
+export interface StakeArgs {
+  value: string;
+  controller: { id: string };
+  payee: 'Stake' | 'Stash' | 'Controller' | { account: string };
+}
+
+export type proxyType =
+  | 'any'
+  | 'nontransfer'
+  | 'governance'
+  | 'staking'
+  | 'unusedsudobalances'
+  | 'identityjudgement'
+  | 'cancelproxy';
+
+export interface AddProxyArgs {
+  delegate: string;
+  delay: string | number;
+  proxyType: proxyType;
+}
+
+export interface ProxyArgs {
+  real: string;
+  forceProxyType: proxyType;
+  call:
+    | string
+    | {
+        callIndex?: string;
+        args?: string;
+      };
+}
+export interface TxMethod {
+  args: TransferArgs | StakeArgs | AddProxyArgs | ProxyArgs;
+  name: 'transferKeepAlive' | 'bond' | 'addProxy' | 'proxy';
+  pallet: string;
+}
+
+export interface DecodedTx extends Omit<DecodedUnsignedTx, 'method'> {
+  method: TxMethod;
+}
+
+export interface SignedTxData {
+  blockHash: string;
```

Comment:
Should be the nonce field added? to prevent double spend.

Reply:
https://github.com/BitGo/BitGoJS/pull/1632/files#diff-dcddee400b8d02f97be108dc204fffa9f4d18cfa56ba41db85eff18a4cc1c6a0R24